http/proxy_query_service.go

```diff
@@ -57,25 +73,39 @@ func (h *ProxyQueryHandler) handlePostQuery(w http.ResponseWriter, r *http.Reque
 		return
 	}
 
+	w.Header().Set("Trailer", QueryStatsTrailer)
+
 	hd, ok := req.Dialect.(HTTPDialect)
 	if !ok {
 		EncodeError(ctx, fmt.Errorf("unsupported dialect over HTTP %T", req.Dialect), w)
 		return
 	}
 	hd.SetHeaders(w)
 
-	n, err := h.ProxyQueryService.Query(ctx, w, &req)
+	cw := iocounter.Writer{Writer: w}
+	stats, err := h.ProxyQueryService.Query(ctx, &cw, &req)
+	n := cw.Count()
```

Comment:
I see this is done to preserve the old code, but it seems unnecessary. You can get rid of line 87 and do `cw.Count() == 0` on line 89. This reduces the scope where this variable could be used which is good for code readability.